### PR TITLE
DCJ-649: Use OidcBroker for sign-out functionality

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-649
* Partially addresses https://broadworkbench.atlassian.net/browse/DCJ-89
* Adds unit tests to start the coverage of `auth.ts`: https://broadworkbench.atlassian.net/browse/DCJ-328
* Adds unit tests to start the coverage of `oidcBroker.ts`: https://broadworkbench.atlassian.net/browse/DCJ-330

### Summary
- This PR tries to carry forward the effort started in https://github.com/DataBiosphere/duos-ui/pull/2611
- This implements a single feature - Sign Out via OIDC.
- `auth.ts` is a minimized version of [`auth.ts`](https://github.com/DataBiosphere/duos-ui/pull/2611/files#diff-12dab7e8b131f803fbdcb63c047768ac2c7c55e0d49d236cf42e13740277e993) in the above PR
- `oidcBroker.ts` is a minimized version of [`oidcBroker.ts`](https://github.com/DataBiosphere/duos-ui/pull/2611/files#diff-c22f0b724cae791bb887d11ce9ada280a1581625e133634e5c2356a5637d6980) in the above PR
- Due to testing, I updated the component test action to store failure screenshots for review.

Follow-on work is defined in a variety of tickets under the [DUOS-2534](https://broadworkbench.atlassian.net/browse/DUOS-2534) epic.

### Testing Notes
Testing should include reviewing the developer panel in a browser when logging into a local instance. 
- Signing in is unchanged so there should be no regressions
- Sign-out mimics the original implementation but additionally uses OidcBroker to invoke OIDC library functionality. This will be expanded in future work that layers in additional sign-in functionality
- Inspecting local session storage should show user info when logged in, and cleared out when signed out.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

[DUOS-2534]: https://broadworkbench.atlassian.net/browse/DUOS-2534?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by fboulnois — APPROVED** [review]
looks good, thanks for adding some tests 👍

File: .github/workflows/component-tests.yml [hunk 1]
```diff
@@ -16,3 +16,14 @@ jobs:
...
           component: true
           browser: chrome
+      - uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: cypress-screenshots
+          path: cypress/screenshots
+          if-no-files-found: ignore # 'warn' or 'error' are also available, defaults to `warn`
+      - uses: actions/upload-artifact@v4
+        with:
```
Comment (s-rubenstein) [inline .github/workflows/component-tests.yml:27]: Should this also have `if: failure()`?

File: cypress/component/Auth/auth.spec.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      url: '/config.json',
+      hostname: 'localhost',
+    }, {'env': 'ci'});
+    cy.stub(OAuth2, 'getConfig').returns({
+      'authorityEndpoint': 'authorityEndpoint',
+      'clientId': 'clientId'
+    });
+    await Auth.initialize();
+  });
+  it('Sign Out Clears the session when called', async function () {
```
Comment (s-rubenstein) [inline cypress/component/Auth/auth.spec.ts:24]: Do you store anything in localStorage on DUOS? That might also be good to check if so.

File: cypress/component/TermsOfService/tos_acceptance.spec.js [hunk 3]
```diff
@@ -1,18 +1,36 @@
...
-import {Storage} from '../../../src/libs/storage';
 import {Navigation} from '../../../src/libs/utils';
+import {OAuth2} from '../../../src/libs/ajax/OAuth2';
+import {ToS} from '../../../src/libs/ajax/ToS';
+import {Storage} from '../../../src/libs/storage';
 
 const text = 'TOS Text';
 const mocks = {
-  history: { push() {} }
+  history: {
```
Comment (s-rubenstein) [inline cypress/component/TermsOfService/tos_acceptance.spec.js:14]: Nit: Is this a lint change? I personally find it more readable the other way.

File: src/App.jsx [hunk 4]
```diff
@@ -66,14 +65,6 @@ function App() {
...
 
-  const signOut = async () => {
-    const clientId = await Config.getGoogleClientId();
-    await GoogleIS.revokeAccessToken(clientId);
-    await Storage.setUserIsLogged(false);
-    await Storage.clearStorage();
-    await setIsLoggedIn(false);
-  };
-
   const signIn = async () => {
```
Comment (s-rubenstein) [inline src/App.jsx:68]: This will move to OIDC with https://broadworkbench.atlassian.net/browse/DCJ-654, correct?

**Review by s-rubenstein — APPROVED** [review]
This looks reasonable - I am curious how it works with sign in using previous auth, but sign out using oidc? Presumably its backwards compatible because Google Auth is included within OIDC?

**Comment by rushtong** [comment]
> This looks reasonable - I am curious how it works with sign in using previous auth, but sign out using oidc? Presumably its backwards compatible because Google Auth is included within OIDC?

Both sign-in and sign-out are fully functional. Sign-in is the bigger lift so I'm working on that now, but separately. The sign-out exercises both the old way of clearing local storage as well as triggering the OIDC library sign-out functionality.

File: src/App.jsx (hunk 4 above)
Comment (rushtong) [reply src/App.jsx:68]: Correct. I'm trying to de-tangle the [sign-in code here](https://github.com/DataBiosphere/duos-ui/pull/2611) so it's less of a boil-the-ocean change.

File: .github/workflows/component-tests.yml (hunk 1 above)
Comment (rushtong) [reply .github/workflows/component-tests.yml:27]: Yes, nice catch!

File: cypress/component/Auth/auth.spec.ts (hunk 2 above)
Comment (rushtong) [reply cypress/component/Auth/auth.spec.ts:24]: We do - I'll boost the uses of `Storage.setXYZ()` and check those too, thank you!

File: cypress/component/TermsOfService/tos_acceptance.spec.js (hunk 3 above)
Comment (rushtong) [reply cypress/component/TermsOfService/tos_acceptance.spec.js:14]: Yes, that's it exactly - I manually unformatted it, then ran our eslint checks and it made the same change.

**Pull request merged**